src/ufuncs.c

```diff
@@ -398,22 +465,26 @@ static void YY_Y_func(char **args, npy_intp *dimensions,
                       npy_intp *steps, void *data)
 {
     FuncGEOS_YY_Y *func = (FuncGEOS_YY_Y *)data;
-    void *context_handle = geos_context[0];
     GEOSGeometry *in1, *in2, *ret_ptr;
 
+    GEOS_INIT;
```

Comment:
I suppose the reason here that those can't be threaded is because of `OUTPUT_Y`? 

Specifically for those binary operations, it would be nice to get those multi-threaded as well (as those can typically be costly). 
I suppose a future improvement could be to do this in two loops (first one to gather the pointers, second one to create the geometry objects). Since I don't think python interaction can be avoided here to achieve a single loop (as you did for `get_geom`)? As we have to manage the reference counts.

Reply:
The geometries cannot be created without the GIL. There is really no way around that, as the python memory management (which allocates memory for the PyObjects) needs the GIL.

The two-step loop is a logical choice, I think we would benefit from that. There might be trade-offs however, I wonder with how many geometries this starts to pay off.

Reply:
The constructive ufuncs like intersection and union are typically expensive functions (and thus where pygeos does not give that much of speedup, as the looping part is small compared to the actual GEOS algo). So I think those could especially benefit from parallelizing. But indeed, would need to check in practice.

Reply:
I like to leave this for a new issue - the PR is already large enough as is